pkg/licensing/normalize.go

```diff
@@ -175,16 +587,84 @@ var pythonLicenseExceptions = map[string]string{
 // 'BSD-3-CLAUSE and GPL-2' => {"BSD-3-CLAUSE", "GPL-2"}
 // 'GPL-1+ or Artistic, and BSD-4-clause-POWERDOG' => {"GPL-1+", "Artistic", "BSD-4-clause-POWERDOG"}
 // 'BSD 3-Clause License or Apache License, Version 2.0' => {"BSD 3-Clause License", "Apache License, Version 2.0"}
-// var LicenseSplitRegexp = regexp.MustCompile("(,?[_ ]+or[_ ]+)|(,?[_ ]+and[_ ])|(,[ ]*)")
-
 var licenseSplitRegexp = regexp.MustCompile("(,?[_ ]+(?:or|and)[_ ]+)|(,[ ]*)")
 
-func Normalize(name string) string {
+// version number match
+var versionRegexpString = "([A-UW-Z)]{2,})( LICENSE)?\\s*[,(-]?\\s*(V|V\\.|VERSION|VERSION-|-)?\\s*([1-9](\\.\\d)*)[)]?"
+
+// case insensitive version match anywhere in string
+var versionRegexp = regexp.MustCompile("(?i)" + versionRegexpString)
+
+// version suffix match
+var versionSuffixRegexp = regexp.MustCompile(versionRegexpString + "$")
```

Comment:
I'm not sure I understand where we need this.
Can you add an example in the comment?

Reply:
I tried to avoid using regex if possible.
Maybe we can avoid using this regex here

Reply:
The many version variations are in normalize_test.

Most of the mappings in current trivy version or in [oss-review-toolkit/ort](https://github.com/oss-review-toolkit/ort/blob/fc5389c2cfd9c8b009794c8a11f5c91321b7a730/utils/spdx/src/main/resources/declared-license-mapping.yml) are because of slight variations in the way the version is declared in the license.

So this regexp allows to:
1. greatly limit the number of mappings
2. avoid missing version mappings

This regexp is strict by checking only version suffixes.